src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/ReferenceTransactionNumber.java

```diff
@@ -1,29 +1,39 @@
 package uk.nhs.digital.nhsconnect.nhais.model.edifact;
 
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
 import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
 
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
 public class ReferenceTransactionNumber extends Reference {
 
-    public ReferenceTransactionNumber() {
-        super("TN", "");
-    }
-
-    public Long getTransactionNumber() {
-        return Long.decode(getReference());
-    }
+    public static final String QUALIFIER = "TN";
 
-    public ReferenceTransactionNumber setTransactionNumber(Long transactionNumber) {
-        this.setReference(Long.toString(transactionNumber));
-        return this;
-    }
+    private @NonNull Long transactionNumber;
 
     @Override
     protected void validateStateful() throws EdifactValidationException {
-        if (getReference() == null) {
-            throw new EdifactValidationException(getKey() + ": Attribute qualifier is required");
-        }
-        if (getReference().isEmpty()) {
+        if (StringUtils.isEmpty(getReference())) {
```

Comment:
As above, we should validate each data item separately. Only whole numbers in the range [1, 9999999] are valid transaction ids.

Reply:
Note that transaction id is "seven nines" while the sequence numbers are "eight nines"